src/trace.rs

```diff
@@ -271,13 +279,72 @@ impl<S> BroadcastLayer<S> {
 
 impl<S> Layer<S> for BroadcastLayer<S>
 where
-    S: Subscriber + 'static,
+    S: Subscriber + 'static + for<'lookup> LookupSpan<'lookup>,
 {
-    fn on_event(&self, event: &Event<'_>, _ctx: Context<'_, S>) {
-        // Try buffering the event, and if we're not buffering anymore, try to send it along via the
-        // trace sender if it's been established.
-        if !try_buffer_event(event) {
-            try_broadcast_event(event);
+    fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
+        if should_process_tracing_event() {
+            let mut log = LogEvent::from(event);
+            // Add span fields if available
+            if let Some(parent_span) = ctx.event_span(event) {
+                for span in parent_span.scope().from_root() {
+                    if let Some(fields) = span.extensions().get::<SpanFields>() {
+                        for (k, v) in &fields.0 {
+                            log.insert(format!("vector.{}", k).as_str(), v.clone());
+                        }
+                    }
+                }
+            }
+            // Try buffering the event, and if we're not buffering anymore, try to
+            // send it along via the trace sender if it's been established.
+            if !try_buffer_event(&log) {
+                try_broadcast_event(log);
+            }
+        }
+    }
+
+    fn on_new_span(
+        &self,
+        attrs: &tracing_core::span::Attributes<'_>,
+        id: &tracing_core::span::Id,
+        ctx: Context<'_, S>,
+    ) {
+        let span = ctx.span(id).expect("span must already exist!");
+        let mut fields = SpanFields::default();
+        attrs.values().record(&mut fields);
+        span.extensions_mut().insert(fields);
+    }
+}
+
+#[derive(Default, Debug)]
+struct SpanFields(HashMap<&'static str, Value>);
+
+impl SpanFields {
+    fn record(&mut self, field: &tracing_core::Field, value: impl Into<Value>) {
+        let name = field.name();
+        if name.starts_with("component_") {
+            self.0.insert(name, value.into());
```

Comment:
A comment here about why we only want these labels might help future readers.

Reply:
Added in [c9df1bd](https://github.com/vectordotdev/vector/pull/12807/commits/c9df1bda0307763cbea89ad4dc1157092e1c06f5)